app/screens/settings/settings.tsx

```diff
@@ -0,0 +1,184 @@
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import SettingOption from './setting_option';
+
+const edges: Edge[] = ['left', 'right'];
+const CLOSE_BUTTON_ID = 'close-settings';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+        },
+        wrapper: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.06),
+            ...Platform.select({
+                ios: {
+                    flex: 1,
+                    paddingTop: 35,
+                },
+            }),
+        },
+        divider: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            height: 1,
+        },
+        middleDivider: {
+            borderTopWidth: 1,
+            borderBottomWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.1),
+            height: 35,
+        },
+        group: {
+            backgroundColor: theme.centerChannelBg,
+        },
+        innerContainerStyle: {
+            paddingLeft: 8,
+        },
+        menuLabel: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200),
+        },
+    };
+});
+
+type SettingsProps = {
+    componentId: string;
+    siteName: string;
+    showHelp: boolean;
+}
+
+//todo: handle display on tablet
+
+const Settings = ({componentId, showHelp, siteName}: SettingsProps) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const styles = getStyleSheet(theme);
+
+    const closeButton = useMemo(() => {
+        return {
+            id: CLOSE_BUTTON_ID,
+            icon: CompassIcon.getImageSourceSync('close', 24, theme.centerChannelColor),
+            testID: CLOSE_BUTTON_ID,
+        };
+    }, [theme.centerChannelColor]);
+
+    const close = useCallback(() => {
+        dismissModal({componentId});
+
+        return true;
+    }, []);
+
+    useEffect(() => {
+        setButtons(componentId, {
+            leftButtons: [closeButton],
+        });
+    }, []);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                if (buttonId === CLOSE_BUTTON_ID) {
+                    close();
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, []);
+
+    const onPressHandler = () => {
+        return Alert.alert(
+            'The functionality you are trying to use has not yet been implemented.',
+        );
+    };
+
+    const goToAbout = preventDoubleTap(() => {
```

Comment:
Shouldn't we use useCallback here?

Reply:
I will profile this whole feature at the end and implement hooks as needed